javascript/code-challenges/challenges-10.test.js

```diff
@@ -0,0 +1,338 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named returnTen, takes in a string and uses split and splice to return the last 10 characters from that string as elements of an array.
+
+------------------------------------------------------------------------------------------------ */
+
+function returnTen(str){
+  let stringSplit = str.split('');
+  let lastTen = stringSplit.splice(stringSplit.length - 10, 10);
+  return lastTen;
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named findMax that takes in a matrix of positive numbers and returns the number with the highest value.
+
+For example:
+[
+  [1, 3, 4, 5],
+  [4, 5, 6],
+  [23, 5, 5]
+]
+
+return: 23
+------------------------------------------------------------------------------------------------ */
+const findMax = (matrix) => {
+  let allNums = [];
+  matrix.forEach(array => {
+    allNums.push(...array);
+  });
+  return Math.max(...allNums);
+};
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named totalSum that takes in a matrix of numbers and returns the totalSum of all the numbers.
+
+For example:
+[
+  [1, 3, 4, 5],
+  [4, 5, 1],
+  [2, 5, 5]
+]
+
+return: 35
+------------------------------------------------------------------------------------------------ */
+const totalSum = (matrix) => {
+  let grandTotal = 0;
+  matrix.forEach(array => {
+    array.forEach(innerArray => {
+      grandTotal += innerArray;
+    });
+  });
+  return grandTotal;
+};
+
```

Comment:
Hi there ! Nice work so far on the challenges :) 

For this one, maybe see if you can use flat() then reduce() 👍  If ya want to stay on the `forEach` path see if you can first use `forEach` then use reduce to get the total sum.

Hopefully this helps!